navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -151,28 +167,110 @@ public void userOffRoute(Location location) {
   @Override
   public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
     if (validRouteResponse(response)) {
-      startNavigation(response.body().getRoutes().get(0));
+      if (navigationRunning) {
+        updateNavigation(response.body().getRoutes().get(0));
+        instructionView.hideRerouteState();
+        summaryBottomSheet.hideRerouteState();
+      } else {
+        startNavigation(response.body().getRoutes().get(0));
+      }
     }
   }
 
   @Override
-  public void onFailure(Call<DirectionsResponse> call, Throwable t) {
-  }
+  public void onFailure(Call<DirectionsResponse> call, Throwable t) {}
 
   private void bind() {
     mapView = findViewById(R.id.mapView);
     instructionView = findViewById(R.id.instructionView);
+    summaryBottomSheet = findViewById(R.id.summaryBottomSheet);
+    cancelBtn = findViewById(R.id.cancelBtn);
+    expandArrow = findViewById(R.id.expandArrow);
+    summaryOptions = findViewById(R.id.summaryOptions);
+    summaryDirections = findViewById(R.id.summaryDirections);
+    directionsOptionLayout = findViewById(R.id.directionsOptionLayout);
+    sheetShadow = findViewById(R.id.sheetShadow);
+  }
+
+  private void initClickListeners() {
+    directionsOptionLayout.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        sheetShadow.setVisibility(View.GONE);
+        summaryOptions.setVisibility(View.GONE);
+        summaryDirections.setVisibility(View.VISIBLE);
+      }
+    });
+    cancelBtn.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        // TODO Finish with cancelled result code
```

Comment:
handled in a different PR?

Reply:
going to hold on this for a later release with the IntentBuilder